logging/middleware.go

```diff
@@ -0,0 +1,7 @@
+package logging
+
+import "net/http"
+
+type Middleware interface {
+	Middleware(next http.Handler) http.Handler
```

Comment:
I do not believe this interface is necessary because the format of the middleware functions are compatible with [http.HandleFunc](https://pkg.go.dev/net/http#HandleFunc).

Reply:
The interface in this case is the function type http.HandleFunc.   I don't believe we need to define any other type in this case.